frontend/src/metabase/query_builder/components/view/ViewFooter/ViewFooter.tsx

```diff
@@ -0,0 +1,100 @@
+import cx from "classnames";
+
+import ButtonBar from "metabase/components/ButtonBar";
+import CS from "metabase/css/core/index.css";
+import { useSelector } from "metabase/lib/redux";
+import { getIsVisualized } from "metabase/query_builder/selectors";
+import * as Lib from "metabase-lib";
+
+import { ViewFooterRoot } from "../ViewFooter.styled";
+
+import {
+  CenterViewFooterButtonGroup,
+  type CenterViewFooterButtonGroupProps,
+} from "./CenterViewFooterButtonGroup";
+import {
+  LeftViewFooterButtonGroup,
+  type LeftViewFooterButtonGroupProps,
+} from "./LeftViewFooterButtonGroup";
+import {
+  RightViewFooterButtonGroup,
+  type RightViewFooterButtonGroupProps,
+} from "./RightViewFooterButtonGroup";
+
+type ViewFooterProps = LeftViewFooterButtonGroupProps &
+  CenterViewFooterButtonGroupProps &
+  RightViewFooterButtonGroupProps;
+
+export const ViewFooter = ({
+  question,
+  result,
+  className,
+  isShowingChartTypeSidebar,
+  isShowingChartSettingsSidebar,
+  isShowingRawTable,
+  onOpenChartType,
+  onCloseChartType,
+  onOpenChartSettings,
+  onCloseChartSettings,
+  setUIControls,
+  isObjectDetail,
+  isTimeseries,
+  visualizationSettings,
+  isShowingTimelineSidebar,
+  onOpenTimelines,
+  onCloseTimelines,
+}: ViewFooterProps) => {
+  const isVisualized = useSelector(getIsVisualized);
```

Comment:
Note: this references the query builder Redux selector, we might have to update it to support the SDK use-case later?

Reply:
yep. *but* the idea of this set of PRs is to split the redux selectors + dispatches to the components that actually use them, instead of drilling everything down. What that means in the future is that we'll be able to isolate the actual component that needs that stuff in the future, rather than the whole sidebar. We'll modularize things and then I can start adding features to the SDK